### PR TITLE
Fifth iteration of the Court pallet

### PR DESCRIPTION
**Pull request by c410-f3r**
depends on #231 
cc #200 

* Updates nightly to a more recent version
* Finishes `on_dispute` implementation

### CODE REVIEW

File: zrml/prediction-markets/src/lib.rs [hunk 1]
```diff
@@ -868,6 +929,17 @@ mod pallet {
...
+    /// been issued.
+    #[pallet::storage]
+    pub type Disputes<T: Config> = StorageMap<
+        _,
+        Blake2_128Concat,
+        MarketIdOf<T>,
+        Vec<MarketDispute<T::AccountId, T::BlockNumber>>,
+        ValueQuery,
+    >;
+
```
Comment (c410-f3r) [inline zrml/prediction-markets/src/lib.rs:942]: Moving back to PM because `Disputes` is used by `court` and `simple-disputes`

**Comment by lsaether** [comment]
#231 was merged, conflicts need to get fixed here

File: primitives/src/constants.rs [hunk 2]
```diff
@@ -25,6 +25,12 @@ parameter_types! {
     pub const MaxReserves: u32 = 50;
 }
 
+// Court
+parameter_types! {
+    pub const CourtCaseDuration: u64 = 2 * BLOCKS_PER_DAY;
```
Comment (lsaether) [inline primitives/src/constants.rs:30]:
```suggestion
    pub const CourtCaseDuration: u64 = BLOCKS_PER_DAY;
```

**Comment by c410-f3r** [comment]
Done

**Review by sea212 — APPROVED** [review]
Thanks, looks good.

File: zrml/court/src/lib.rs [hunk 3]
```diff
@@ -46,25 +51,44 @@ mod pallet {
...
+            if Jurors::<T>::get(&who).is_some() {
                 return Err(Error::<T>::JurorAlreadyExists.into());
             }
             let jurors_num = Jurors::<T>::iter().count();
             let jurors_num_plus_one = jurors_num.checked_add(1).ok_or(ArithmeticError::Overflow)?;
             let stake = Self::current_required_stake(jurors_num_plus_one);
-            Jurors::<T>::insert(&account_id, Juror { staked: stake, status: JurorStatus::Ok });
-            CurrencyOf::<T>::reserve(&account_id, stake)?;
+            Jurors::<T>::insert(&who, Juror { staked: stake, status: JurorStatus::Ok });
+            CurrencyOf::<T>::reserve(&who, stake)?;
```
[the remark this answers is not in the document]
Comment (c410-f3r) [reply zrml/court/src/lib.rs:72]: Done

File: zrml/court/src/lib.rs [hunk 4]
```diff
@@ -46,25 +51,44 @@ mod pallet {
...
+            Ok(())
+        }
+
+        #[frame_support::transactional]
```
[the remark this answers is not in the document]
Comment (c410-f3r) [reply zrml/court/src/lib.rs:76]: Removed

**Review by lsaether — APPROVED** [review]
Thanks!

**Comment by c410-f3r** [comment]
Thank you guys

**Pull request merged**